go/plugins/postgresql/docstore.go

```diff
@@ -275,3 +275,101 @@ func (ds *docStore) generateAddDocumentsQuery(id, content, embedding string, met
 	query := insertStmt + valuesStmt
 	return query, values, nil
 }
+
+/* +++++++++++++++++++++++
+		Advance methods
+ +++++++++++++++++++++++++ */
+
+// ApplyVectorIndex creates an index in the table of the embeddings.
+func (ds *docStore) ApplyVectorIndex(ctx context.Context, index BaseIndex, name string, concurrently bool) error {
```

Comment:
Docstore isn't accessible by developers. The receiver struct should be the retriever